src/versionbits.cpp

```diff
@@ -27,6 +27,12 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex*
     int64_t nTimeStart = BeginTime(params);
     int64_t nTimeTimeout = EndTime(params);
 
+    // Check if this deployment has a fixed activation height.
+    if (nTimeStart < 0) {
```

Comment:
Possible to test this logic in versionbits_tests.cpp?

Reply:
Agree, this really needs a test :(

Might be better to do it in the python framework though, to make sure that negative values are parsed correctly all the way down from the cmdline.

Reply:
I've added some tests to versionbits_tests in https://github.com/sipa/bitcoin/pull/106 fwiw.

Reply:
@ajtowns Thanks, I've included some of those commits, but reworked them a bit (I've simplied the code to only support always-active, so the tests are easier now).